src/indexer/segment_updater.rs

```diff
@@ -573,35 +585,42 @@ impl SegmentUpdater {
     fn end_merge(
         &self,
         merge_operation: MergeOperation,
-        mut after_merge_segment_entry: SegmentEntry,
-    ) -> crate::Result<SegmentMeta> {
+        mut after_merge_segment_entry: Option<SegmentEntry>,
+    ) -> crate::Result<Option<SegmentMeta>> {
         let segment_updater = self.clone();
-        let after_merge_segment_meta = after_merge_segment_entry.meta().clone();
+        let after_merge_segment_meta = after_merge_segment_entry
```

Comment:
after_merge_segment_meta is not used below...

Reply:
yes it's returned